File: backend/internal/api/handler/creator.go
package handler

import (
	"net/http"

	"github.com/gin-gonic/gin"

	"insta360-insight/internal/model"
	"insta360-insight/internal/service"
)

// Creator 达人分析 HTTP 处理器
type Creator struct {
	svc *service.CreatorService
}

func NewCreator(svc *service.CreatorService) *Creator {
	return &Creator{svc: svc}
}

// Kpi GET /api/creator/kpi
func (h *Creator) Kpi(c *gin.Context) {
	data, err := h.svc.Kpi(c.Request.Context(), bindFilter(c))
	if err != nil {
		fail(c, err)
		return
	}
	c.JSON(http.StatusOK, model.OK(data))
}

// Trend GET /api/creator/trend
func (h *Creator) Trend(c *gin.Context) {
	data, err := h.svc.Trend(c.Request.Context(), bindFilter(c))
	if err != nil {
		fail(c, err)
		return
	}
	c.JSON(http.StatusOK, model.OK(data))
}

// Platforms GET /api/creator/platforms
func (h *Creator) Platforms(c *gin.Context) {
	data, err := h.svc.Platforms(c.Request.Context(), bindFilter(c))
	if err != nil {
		fail(c, err)
		return
	}
	c.JSON(http.StatusOK, model.OK(data))
}

// Tracks GET /api/creator/tracks
func (h *Creator) Tracks(c *gin.Context) {
	data, err := h.svc.Tracks(c.Request.Context(), bindFilter(c))
	if err != nil {
		fail(c, err)
		return
	}
	c.JSON(http.StatusOK, model.OK(data))
}

// Audience GET /api/creator/audience
func (h *Creator) Audience(c *gin.Context) {
	data, err := h.svc.Audience(c.Request.Context(), bindFilter(c))
	if err != nil {
		fail(c, err)
		return
	}
	c.JSON(http.StatusOK, model.OK(data))
}

// List GET /api/creator/list
func (h *Creator) List(c *gin.Context) {
	data, err := h.svc.List(c.Request.Context(), bindFilter(c))
	if err != nil {
		fail(c, err)
		return
	}
	c.JSON(http.StatusOK, model.OK(data))
}

File: frontend/src/api/creator.js
// 达人分析相关 API, 全部走 /api 代理(5173 -> 8080)
// 注意: 响应拦截器返回的是完整 { code, data } 信封, 这里统一拆出 data,
// 使 store 拿到的就是业务数组/对象(与 store 内部的数组校验一致)。
import request from './request'

const creator = {
  kpi:       (q) => request.get('/creator/kpi', { params: q }).then(r => r.data),
  trend:     (q) => request.get('/creator/trend', { params: q }).then(r => r.data),
  platforms: (q) => request.get('/creator/platforms', { params: q }).then(r => r.data),
  tracks:    (q) => request.get('/creator/tracks', { params: q }).then(r => r.data),
  audience:  (q) => request.get('/creator/audience', { params: q }).then(r => r.data),
  list:      (q) => request.get('/creator/list', { params: q }).then(r => r.data)
}

export default creator
